PacketDotNet/UdpPacket.cs

```diff
@@ -25,6 +25,14 @@ namespace PacketDotNet
     /// </summary>
     public sealed class UdpPacket : TransportPacket
     {
+        /// <summary>
+        /// Callback function for UdpPacket Payload decoding.
+        /// First parameter is the payload, second parameter if the UdpPacket itself
+        /// returned value is the decoded payload as Packet or null if payload is invalid or not supported
+        /// then internal decoding will continue
+        /// </summary>
+        public static Func<ByteArraySegment, UdpPacket, Packet> UdpPayloadCustomDecoderFunc;
```

Comment:
I think we can move this to TransportPacket (and TransportPacket as second parameter) so TcpPacket can also use it. The result should be a PacketOrByteArraySegment to also make it possible to return a byte array segment if it's needed. 
The name can be CustomPayloadDecoder.

Reply:
Sure I can, but this would contradict a little bit the stuff done packet-connection isn't it?

Reply:
It's a separate project and this would also be useful for Tcp payload parsing 🙂